sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/ConfigurationSettingsFilter.cs

```diff
@@ -17,20 +17,29 @@ public ConfigurationSettingsFilter(string key)
             Argument.AssertNotNull(key, nameof(key));
 
             Key = key;
+            Tags = new ChangeTrackingList<string>();
         }
 
         /// <summary> Initializes a new instance of KeyValueFilter. </summary>
         /// <param name="key"> Filters key-values by their key field. </param>
         /// <param name="label"> Filters key-values by their label field. </param>
-        internal ConfigurationSettingsFilter(string key, string label)
+        /// <param name="tags"> Filters key-values by their tags field.
+        /// Each tag in the list should be expressed as a string in the format `tag=value`.
+        /// </param>
+        internal ConfigurationSettingsFilter(string key, string label, IList<string> tags)
         {
             Key = key;
             Label = label;
+            Tags = tags;
         }
 
         /// <summary> Filters key-values by their key field. </summary>
         public string Key { get; set; }
         /// <summary> Filters key-values by their label field. </summary>
         public string Label { get; set; }
+        /// <summary> Filters key-values by their tags field.
+        /// Each tag in the list should be expressed as a string in the format `tag=value`.
+        /// </summary>
+        public IList<string> Tags { get; set; }
```

Comment:
do we need this property settable?

Reply:
Looks like the other two filter properties `Key` and `Label` are settable, and the public ctor for this class only accepts a `key`. Are you suggesting I add an additional public ctor that accepts the filters so we can make these properties immutable?

Reply:
No, I was suggesting users simply call Tags.Add(...) to add tags. We would need to make the collection allocation lazy though. Maybe it's not worth it if it's too much work

Reply:
+1 to Krzysztof's feedback here.  `Key` and `Label` are not a direct parallel because they're strings and therefore must be directly settable.

By convention, all of the collection types on models across libraries are get-only.    The general interaction pattern is:

```csharp
// Initializer syntax
var filter = new ConfigurationSettingsFilter
{
    Key = "somekey",
    Tags = { "one", "two" }
};

// Mutation
var filter = new ConfigurationSettingsFilter();

filter.Key = "someKey";
filter.Tags.Add("one");
filter.Tags.Add("two");
```

Neither of these requires the collection to be settable on the model.   Something like the following would be my suggestion:

```csharp
// Direct allocation.   Used by most model types in the Azure SDK packages
public IList<string> Tags { get; } = new List<string>();

// Lazy initialization
IList<string> _tags = null;

public IList<string> Tags
{
    get
    {
         _tags ??= new List<string>();
        return _tags;
    }
}
```

The downside to the direct approach is that it performs an allocation for a collection that may not be used.   The downside to the second is that it is not thread-safe and adds an additional conditional check to every accessor use.

Either way is fine.  This isn't a hot path that we need to microbenchmark and optimize for.   We're in a scenario where this type is used to make network calls, which are going to dominate the performance metrics and are already allocation heavy.  We also have no guarantees around our model types and thread-safety.  

I prefer approach #1, personally, but its your call.  😄

Reply:
@jsquire  - thanks for the explanation. I removed the setters in that case. For simplicity, I went with just initializing the list directly. Though the public ctors for both `ConfigurationSettingsFilter` and `SettingSelector` do not take in tags as a parameter, and since there is no setter then users can't initialize the tags when creating the filter objects. The [ConfigurationSetting](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/ConfigurationSetting.cs#L113)  model seems to workaround this by having an internal setter on its' tags. If this is the convention, then I can make the same change.. Do let me know your thoughts, thanks.

Reply:
> since there is no setter then users can't initialize the tags when creating the filter objects.

So long as the getter exists, both the initialization syntax and mutation approaches above work.   The initializer syntax is sugar for mutation, though it looks like an assignment.    

> The [ConfigurationSetting](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/ConfigurationSetting.cs#L113) model seems to workaround this by having an internal setter on its' tags.

This doesn't enable any developer-facing scenarios but allows library code to pass a complete set.   I don't know why this was done without tracing the code.   If we don't have a scenario where we know we need it, I don't see a reason to add it.